app/controllers/admin/partners_controller.rb

```diff
@@ -165,6 +164,18 @@ def setup
 
     private
 
+    def validate_category_limit
```

Comment:
Lordy, remove this!

1. validations should only happen in the model (one place for data business logic)
2. the rule here is checking for a maximum of 5 and the model says it should be 3 (which is correct?)
3. this only happens on create, but not update (different behaviours depending on what the user is doing)
4. it also stops all the other validations running so if there are any other problems with the other fields they will not be shown.